core/src/main/java/io/questdb/griffin/engine/table/SelectedRecordCursorFactory.java

```diff
@@ -151,11 +167,20 @@ public boolean recordCursorSupportsRandomAccess() {
         return base.recordCursorSupportsRandomAccess();
     }
 
+    public void replaceBaseFactory(RecordCursorFactory base) {
```

Comment:
nit: some kind of warning on unsafety of this method (metadata must match) would be nice to have here